Add transparency feature to google calendar component.

cgtobi opened the pull request:
## Description:
This implements the transparency feature for google calendars. If for example an all-day event is present but it is set to 'free' HA now skips those events and shows the next. This can be turned on/off per calendar.

The yaml config looks like this:

```yaml
- cal_id: calendar@googlemail.com
  entities:
  - device_id: calendar
    name: Calendar
    track: true
    transparency: true
```

transparency is optional and defaults to false. It enables/disables whether to respect/ignore the transparency tag which can be opaque (busy) or transparent (free).

**Related issue (if applicable):** fixes #6076 (only partial)

**Pull request in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io) with documentation (if applicable):** home-assistant/home-assistant.github.io#5101

## Example entry for `configuration.yaml` (if applicable):
```yaml

```

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io)

If the code communicates with devices, web services, or third-party tools:
  - [ ] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [ ] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [ ] New dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [ ] New files were added to `.coveragerc`.

If the code does not interact with devices:
  - [ ] Tests have been added to verify that the new code works.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L14
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L54

cgtobi committed:
Remove len call.

cgtobi committed:
Remove len call.

cgtobi committed:
Add transparency feature for google calendars.

cgtobi committed:
Add transparency feature for google calendars.

tinloaf commented on homeassistant/components/calendar/google.py:
```diff
@@ -18,11 +18,11 @@
 }
 
-MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=15)
+MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=1)
```
Were the 15 minutes previously chosen with some deeper reason in mind? Some API rate limit from Google?

cgtobi replied:
Oh, I forgot to set that back to 15. Thanks for the hint.

tinloaf commented on homeassistant/components/calendar/google.py:
```diff
@@ -45,18 +45,21 @@ class GoogleCalendarEventDevice(CalendarEventDevice):
 
-    def __init__(self, calendar_service, calendar_id, search=None):
+    def __init__(self, calendar_service, calendar_id, search=None,
+                 transparency=True):
```
Setting a default for `transparency` here doesn't make sense if you're defaulting to `None` above, does it?

cgtobi replied:
Well spotted and thanks again. I'll fix it in a minute.

tinloaf reviewed (commented):
Aside from the inline comments - is "transparency" the official google calendar terminology? I found the term rather confusing, and thought it was some visual thing.

cgtobi commented:
The terminology comes from google calendar API reference.

`transparency `

Whether the event blocks time on the calendar. Optional. Possible values are:
- "opaque" - Default value. The event does block time on the calendar. This is equivalent to setting Show me as to Busy in the Calendar UI.
- "transparent" - The event does not block time on the calendar. This is equivalent to setting Show me as to Available in the Calendar UI.

See: https://developers.google.com/calendar/v3/reference/events

cgtobi commented:
Should the default value be `false` instead of `true`? That way it would be less 'breaking'.

cgtobi committed:
Remove double default value assignment.

cgtobi committed:
Remove double default value assignment.

balloob committed:
Always enable config entries & remove config_entry_example (#13663)

emlove committed:
Add missing service docs for hs_color (#13667)

balloob committed:
Bump frontend to 20180404.0

mountainsandcode committed:
Add regex functions as templating helpers (#13631)

* Add regex functions as templating helpers

* Add regex functions as templating helpers - Style fixed

* Templating filters, third time lucky?

stephanerosi committed:
Channel up and down for webostv (#13624)

5LICK committed:
Added headers configuration variable to notify.rest component (#13674)

* Added headers configuration variable to notify.rest component

* Fix code style

scop committed:
Spelling fixes (#13681)

syssi committed:
Xiaomi MiIO Light: White Philips Candle Light support (#13682)

cdce8p committed:
Bugfixes HomeKit covers, lights (#13689)

* covers -> current_position attribute
* lights -> hue and saturation attribute

cdce8p committed:
Minor style changes, cleanup (#13654)

* Minor style changes, cleanup
* Change 'self._entity.id' to 'self.entity_id'
* Use const 'STATE_OFF'
* Added CATEGORY constants
* Removed *args from accessory types
* Changed 'self._hass' to 'self.hass'
* Added log debug msg (for added lights)

turbokongen committed:
Bugfix: Zwave Print_node to logfile instead of console (#13302)

* Print to logfile instead of console

* Review changes

* Typo

cdce8p committed:
Update HAP-python to 1.1.8 (#13563)

* Bump version to HAP-python==1.1.8
* Required changes for version change
* Small bugfix lights

danielperna84 committed:
Update ha-philips_js to 0.0.3 (#13702)

* Update requirements_all.txt

* Update philips_js.py

kellerza committed:
Coverage & Codeowners (#13700)

teharris1 committed:
Bump insteonplm to 0.8.6 to fix sensor message handling (#13691)

shuaiger committed:
Fix asuswrt ap mode failure (#13693)

* fix asuswrt ap mode failure

When using ap mode, asuswrt device_tracker does dont work properly as ip can not be retrieved from wl command. This version fixed the issue.

* save 1 line code

* another 2 lines saved

* typo correction

morberg committed:
Support color temperature in Homekit (#13658)

* Add support for color temperature
* Add test for color temp

robmarkcole committed:
Remove unused CONF_WATCHERS (#13678)

`CONF_WATCHERS` was from an earlier version, now unused

tadly committed:
Add media type separation for video/movie (#13612)

* added media type separation for video/movie

* updated all media_player components to reflect new media type

PlanetJ committed:
Adding configration to disable ip address as a requirement Fixes: #13399 (#13692)

* Adding configration to disable ip address as a requirement Fixes: #13399

* Remove whitespace

ikucuze committed:
Tahoma switches (#13636)

* Added the ability to switch Tahoma Garage door relay.

Those are special switches that can only be pushed.
Their state is always OFF, they react to the turn_on action, perform it, but stay OFF

* fixing indents and so on

* CI fix

emlove committed:
Send XY color for non-osram hue bulbs (#13665)

* Send XY color for Philips hue bulbs

* Review fixes

* Comment tweaks

MisterWil committed:
Update AbodePy version to 0.12.3 (#13709)

jmtatsch committed:
Enable autodiscovery for mqtt cameras (#13697)

* Enable autodiscovery for mqtt cameras, BREAKING CHANGE: homogenisation topic->state_topic

* fix line too long

* fix topic->state_topic in test

* image shall not be the state of entity

cgtobi commented:
I wanted to change it to ‘ignore_availability‘ but my internet connection is down for days.

cgtobi committed:
Change keyword to be more explicit.

cgtobi committed:
Change keyword to be more explicit.

cgtobi commented:
Sorry, somehow I messed up big. I'm going to open a new PR.
